components/toggle_modal_button_redux/toggle_modal_button_redux.jsx

```diff
@@ -55,6 +61,16 @@ class ModalToggleButtonRedux extends React.PureComponent {
 
         const ariaLabel = formatMessage({id: 'accessibility.button.dialog', defaultMessage: '{dialogName} dialog'}, {dialogName: props.accessibilityLabel});
 
+        let badge = null;
+        if (this.props.modalId === ModalIdentifiers.PLUGIN_MARKETPLACE) {
+            const firstAdminVisitMarketplaceStatus = getFirstAdminVisitMarketplaceStatus(store.getState());
```

Comment:
I'm a bit surprised to that logic here. Does it make sense to add a `Prop` e.g. `Unread` to `ModalToggleButtonRedux` and pass the value of `getFirstAdminVisitMarketplaceStatus` as such?